### PR TITLE
Convert `examples` to conform to best practices

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
I have updated some of our examples to conform to our best practices - :heavy_check_mark: is done - :hamburger: is not done and should be done in the next few days - airflow currently does not work with our best practices structure. 

:heavy_check_mark: add_your_own 
:heavy_check_mark: airflow_orchestration  
:heavy_check_mark: cloud_secrets_manager  
:heavy_check_mark: evidently_drift_detection   
:heavy_check_mark: facets_visualize_statistics    
:heavy_check_mark: feast_feature_store 
:heavy_check_mark: github_actions_orchestration
:heavy_check_mark: :  huggingface 
:heavy_check_mark: kubeflow_pipelines_orchestration                     
:heavy_check_mark: lightgbm                            
:heavy_check_mark: mlflow_deployment     
:heavy_check_mark: mlflow_tracking    
:heavy_check_mark: neural_prophet         
:heavy_check_mark: pytorch                   
:heavy_check_mark: quickstart    
:heavy_check_mark: scipy          
:cactus: seldon_deployment   - done - Need to be tested with AWS cluster where seldon is installed
:heavy_check_mark: slack_alert           
:heavy_check_mark: step_operator_remote_training     
:heavy_check_mark: vertex_ai_orchestration    
:heavy_check_mark: wandb_tracking           
:heavy_check_mark: whylogs_data_profiling    
:heavy_check_mark: xgboost

https://docs.zenml.io/links-and-resources/resources/best-practices

Besides the obvious many changes to the example structure there have been a few changes in the core repo. The pipeline-run code has moved out of the cli into the pipelines module to be callable from code without invoking subprocess calls or cli-runners.

On top of that the source utils have been adjusted to make sure that `zenml pipeline run ... -c config.yaml` works for our recommended repo structure. Special attention should be payed to the changes in src/zenml/utils/source_utils.py during code review (:eyes: @schustmi  :wink: )

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by htahir1** [comment]
The fact that ✅ is the emoji for not done is the biggest troll today! :D

**Comment by AlexejPenner** [comment]
> The fact that white_check_mark is the emoji for not done is the biggest troll today! :D

Changed it to :hamburger: to motivate volunteers

**Comment by strickvl** [comment]
@fa9r https://docs.zenml.io/links-and-resources/resources/best-practices is the best practices link, but you probably would do better just to look at the model of one of the complete ones. I was going to write up instructions on this process on Friday but didn't get around to it. Maybe overkill given how many we have left and that this is a one-off process. But lmk if you have any questions. I didn't find it super intuitive to do the first one.

**Review by strickvl — COMMENTED** [review]
This is my favourite PR and I am writing my best ever review.

**Comment by fa9r** [comment]
Just did pytorch, kubeflow, airflow. Now only huggingface left, @AlexejPenner I saw you worked on that before, might make sense to refactor that one yourself :D

**Comment by fa9r** [comment]
We're getting close, really looking forward to finally have this merged! 😄

**Comment by AlexejPenner** [comment]
Just some tests missing now :)

**Comment by safoinme** [comment]
Only Seldon's example is left for the test I think. tried to make sure it's right but still would be better to run it if you guys can do that with the AWS Stack @AlexejPenner  @fa9r

**Comment by safoinme** [comment]
Somehow not all examples are really following the same exact format! there are examples where each step or pipeline is inside a separated folder and some are gathered together, should it be unified @AlexejPenner  @strickvl ?

**Comment by strickvl** [comment]
@safoinme like which examples?

**Comment by safoinme** [comment]
- examples where all steps are gathered together: kubeflow_pipeline_orchestration, PyTorch, quickstart, slack_alerter
- examples where each step is in a separated folder: mlflow_tracking,  neural_prophet, Scipy, seldon_deployment, step_operator, Vertex_ai, wandb, why_logs, xgboost, 
@strickvl

File: src/zenml/utils/source_utils.py [hunk 1]
```diff
@@ -523,47 +525,52 @@ def load_source_path_class(
...
-        os.path.relpath(file_path, os.getcwd())
-    )[0].replace(os.path.sep, ".")
+    module_name = os.path.splitext(os.path.relpath(module_path, os.getcwd()))[
+        0
+    ].replace(os.path.sep, ".")
 
-    if full_module_path not in sys.modules:
-        with prepend_python_path(os.path.dirname(file_path)):
+    if module_name in sys.modules:
+        del sys.modules[module_name]
```
Comment (schustmi) [inline src/zenml/utils/source_utils.py:560]: Can you maybe add a short (or long) comment here that explains in detail why the module needs to be deleted, as that is a very strange operation if someone comes access this code.

**Review by schustmi — CHANGES_REQUESTED** [review]

**Comment by fa9r** [comment]
> I know I'm late to this party, but I just had to go through the process of refactoring one of my examples to conform to these best practices and I have to object to the python module hierarchy recommendation. Do we really need to have a separate python sub-module for every single pipeline and step ? I find it unnecessary and difficult to work with.
> 
> A better/simpler hierarchy would be e.g.:
> 
> ```
> example 
>   /pipelines   
>     __init__.py
>     training.py
>     inference.py
>   /steps
>     __init__.py
>     importer.py
>     splitter.py
>     deployer.py
> ...
> ```

I agree 100%. I did all my refactorings like this, which are the "inconsistencies" between the different examples we talked about 😄 For now, I think we agreed that both structures are acceptable since this design choice doesn't affect testing.

**Pull request merged**